### PR TITLE
Updating with endpoint for Article ID search, error handling,  and tests

### PR DESCRIPTION
**Pull request by bayardt**

### CODE REVIEW

File: __tests__/endpoints.test.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            votes: expect.any(Number),
+          });
+        });
+    });
+    test("status: 404 - responds with a 404 if route does not exist.", () => {
+      return request(app)
+        .get("/api/articles/10000")
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe("No article found for article_id: 10000");
```
Comment (daviidbartlett) [inline __tests__/endpoints.test.js:49]: this is not an invalid route

File: app.js [hunk 2]
```diff
@@ -1,13 +1,28 @@
...
+  if (err.status && err.msg) {
+    res.status(err.status).send({ msg: err.msg });
+  }
+  if (err.code === "22P02") {
+    res.status(400).send({ msg: "Bad Request" });
+  } else {
+    res.status(500).send({ msg: "Internal Server Error" });
+    console.log(err);
+  }
+});
```
Comment (daviidbartlett) [inline app.js]: consider breaking this down into separate error handling middleware functions

File: __tests__/endpoints.test.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        .then(({ body }) => {
+          expect(body.msg).toBe("Error 404 - Route not found");
+        });
+    });
+  });
+});
+
+describe("/api/articles/:article_id", () => {
+  describe("GET", () => {
+    test("status: 200 - responds with the specified article.", () => {
```
Comment (popatre) [inline __tests__/endpoints.test.js:28]: Your declaring that you're testing for a specified article, but the test just checks for ANY object that matches those matchers. 

If you're after a specific article, you can out the data you expect to get back from that article. 
At the moment you could be getting the wrong article, and your test wouldnt show this.

**Review by popatre — CHANGES_REQUESTED** [review]

**Review by popatre — APPROVED** [review]
Much better - you're actually checking that the correct data is returning now.

**Pull request merged**